File: backend/src/main/java/com/greenhouse/entity/Plot.java
package com.greenhouse.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 地块实体类
 * 对应表：plots
 */
@Data
@TableName("plots")
public class Plot {
    /**
     * 主键ID
     */
    @TableId(type = IdType.AUTO)
    private Integer id;

    /**
     * 地块编号（1-4）
     */
    private Integer plotNumber;

    /**
     * 地块名称（可选）
     */
    private String name;

    /**
     * 状态（0禁用/1启用）
     */
    private Boolean status;

    /**
     * 创建时间
     */
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    private LocalDateTime updatedAt;
}

File: utio/models/Plot.js
/**
 * 地块模型
 * 对应表：plots
 */

class Plot {
  constructor(data = {}) {
    this.id = data.id || null
    this.plotNumber = data.plot_number || data.plotNumber || null
    this.name = data.name || null
    this.status = data.status !== undefined ? data.status : 1
    this.createdAt = data.created_at || data.createdAt || null
    this.updatedAt = data.updated_at || data.updatedAt || null
  }

  /**
   * 转换为数据库字段格式
   */
  toDbFormat() {
    return {
      id: this.id,
      plot_number: this.plotNumber,
      name: this.name,
      status: this.status ? 1 : 0,
    }
  }

  /**
   * 转换为前端格式
   */
  toFrontendFormat() {
    return {
      id: this.id,
      plotNumber: this.plotNumber,
      name: this.name,
      status: this.status,
      createdAt: this.createdAt,
      updatedAt: this.updatedAt,
    }
  }

  /**
   * 验证数据有效性
   */
  validate() {
    const errors = []
    if (!this.plotNumber || this.plotNumber < 1) errors.push('地块编号必须大于0')
    return errors
  }
}

module.exports = Plot

